internal/reporters/reporter.go

```diff
@@ -0,0 +1,70 @@
+package reporters
+
+import (
+	"reflect"
+	"soarca/logger"
+	"soarca/models/cacao"
+)
+
+type Empty struct{}
+
+var component = reflect.TypeOf(Empty{}).PkgPath()
+var log *logger.Log
+
+func init() {
+	log = logger.Logger(component, logger.Info, "", logger.Json)
+}
+
+// Reporter interfaces
+
+type IWorkflowReporter interface {
+	ReportWorkflow(workflow cacao.Workflow) error
```

Comment:
Should this return an error? Because we don't want to restrict execution I think.

Reply:
If an error is returned, then in the decomposer or executor I just log the error as warning

Reply:
Lets discuss what is the best solution